zerver/views/realm_playgrounds.py

```diff
@@ -0,0 +1,27 @@
+from django.core.exceptions import ValidationError
+from django.http import HttpRequest, HttpResponse
+
+from zerver.decorator import require_realm_admin
+from zerver.lib.actions import do_add_realm_playground
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_error, json_success
+from zerver.models import UserProfile
+
+
+@require_realm_admin
+@has_request_variables
+def add_realm_playground(request: HttpRequest, user_profile: UserProfile,
+                         name: str=REQ(), pygments_language: str=REQ(),
+                         url_prefix: str=REQ()) -> HttpResponse:
```

Comment:
We generally prefer to do validation here, in the REQ parameters, rather than via a try/except with model fields (even if the latter is useful secondary validation to ensure we don't save anything problematic to the database)

Reply:
Quick doubt here, I know you mentioned doing the validation inside REQ parameters,  but for the regex and url cases, we want to do some custom validation (beyond those offered by `lib/validators`. I'm thinking of adding check functions like this in the view with the functions themselves in `lib/playground.py`. 
```
def add_realm_playground(request: HttpRequest, user_profile: UserProfile,
                         name: str=REQ(), pygments_language: str=REQ(),
                         url_prefix: str=REQ()) -> HttpResponse:
    check_valid_pygments_language(pygments_language)
    check_valid_playground_url(url_prefix)
```

Reply:
Yeah, that's the way to do it if `validators.py` doesn't support the validation we need.

(Well, or we could write a custom validator in the style of `validators.py`).